### ISSUE TITLE
Need a publishedBy property

### ISSUE THREAD

**Issue by jeannekitchens** [CredentialEngine/Schema-Development#661]
Publishers of pathways, competencies, and quantitative data can be organizations that are not quality assurance or credentialing organizations that own or offer credentials.  A publishedBy property is needed, as well as another organization class (the class is handled with another issue).  This issue to is only address the publishedBy.

**Comment by siuc-nate** [comment]
What would the definition of such a property be?  How would it be used differently from ownedBy and offeredBy?

**Comment by philbarker** [comment]
The schema.org equivalent is [`publisher`](https://schema.org/publisher) "The publisher of the creative work" :-).  Dublin Core is a bit less circular, but perhaps at the cost of making the difference between publisher and provider (offeredBy) unclear: [`publisher`](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#http://purl.org/dc/terms/publisher) An entity responsible for making the resource available.

A resource owner can licence another agent to publish that resource; once published instances of it can be provided by a third agent. For example,  Lord of the Rings is owned by [Middle-earth Enterprises](https://en.wikipedia.org/wiki/Middle-earth_Enterprises); is published by HarperCollins; is provided by Amazon.

A shot at a definition: 
**`publishedBy`**: an agent that with the permission of the rights owner makes instances of a resource available so that access to them can be offered.

**Comment by stuartasutton** [comment]
+1

**Comment by siuc-nate** [comment]
I think [`offeredBy`](https://credreg.net/ctdl/terms/offeredBy) covers this - I don't see what the gap between the two would be, either in definition or in practical use.

And again, what was the original source/context/need that prompted this? We may find more answers there.

**Comment by philbarker** [comment]
I think the distinction is necessary when you think of learning opportunities and assessments as well as credentials.

An example: SQA publish a qualification, [HND Acting and Performance](https://www.sqa.org.uk/sqa/41445.html), Edinburgh College (and others around Scotland) offer a [course](http://www.edinburghcollege.ac.uk/courses/Performing-Arts/Acting%20and%20Performance%20HND/CR1HDAPB20) that leads to that qualification

I beleive GED operates similarly. IIUC, GED is owned by ACE, the tests are published by Pearson and offered by many independent centres.

You could use offer for both functions, then how would someone know which to go to in order to earn the qualification?

**Comment by siuc-nate** [comment]
In the first example, since SQA owns the data, you would reference them via `ownedBy` and then reference the colleges that offer that credential via `offeredBy`.

[GEDs are a little bit of a special case](https://github.com/CredentialEngine/Schema-Development/issues/609) but in our usage/context are considered to be just another type of Credential in parallel with a High School Diploma. GEDs tend to be reliant on exams, so going with your example's organizations, it would look like this:

GED **Credential**
- owned by ACE
- offered by Pearson

GED **Assessment** (ie GED Credential -> requires -> GED Assessment (Profile))
- owned by ACE and Pearson (the [website](https://home.pearsonvue.com/About-Pearson-VUE/Company-information/Pearson-VUE-companies/GED-Testing-Service.aspx) lists it as a joint venture between them)
- offered by many independent testing centers

In other words, you would go to one of the testing centers to take the GED test that's required for the GED credential.

**Comment by philbarker** [comment]
I do not think that SQA do anything different with their HNDs to what Pearson do with GEDs.  There is a difference between how a University offers the credentials/courses/assessment it owns without publishing them, and how other organizations publish the credentials/courses/assessments they (or other people) own without offering them. But whether offeredBy can adequately cover both scenarios depends on what specifics @jeannekitchens had in mind when she raised this issue.

**Comment by siuc-nate** [comment]
+1 for needing the context.  

I still don't understand what it actually means (in practical terms) to "publish" something intangible like a credential, unless you are referring to publishing it to the registry, which I'm not sure is the case (and is handled in other ways anyway).

I'm also not sure how these would actually be used by the organizations that publish (to the registry) or consume that data, or by end-users.  It feels like splitting hairs in a way that may not be sufficiently beneficial given that it will add greater complexity to the schema (ie connections between organizations and other things), to the documentation (explaining the differences between publisher/offerer/owner to the non-technical customers we serve is not something I could do at this point), and to the description set queries (adding a new branch of connections that have to be checked will make the already-slow queries even slower).

In other words: semantics aside, what is the real value-add of doing this?

**Comment by jeannekitchens** [comment]
This issue is only meant to be about organizations that can/will legitimately publish to the Registry and the web information that they do not directly offer.  Concrete examples include:
1. State agencies publishing education and career pathways.  The state does not offer or own any of the pathways.  They only publish information about them.
2. Associations publishing education and career pathways. The association does not offer or own any of the pathways.  They only publish information about them.
3. State or federal agencies publishing employment and earnings information about credentials, learning opps, or assessments.

Reading the responses, I can't tell if the conversation resolved this or not.  If we base the use of [offeredBy](https://credreg.net/ctdl/terms#offeredBy) and [ownBy](https://credreg.net/ctdl/terms#ownedBy) on their definitions...or any of the other "Bys" do we need a new "By" property or can we close this issue?

**Comment by siuc-nate** [comment]
If I recall correctly, the registry envelope already contains this information (meta information about who put the record into the registry). @mparsons-ce can you point out which envelope properties contain this? 

Anyway, yes, I think this issue is already taken care of.

**Comment by mparsons-ce** [comment]
There is an envelope property for publisher and secondary publisher:
![image](https://user-images.githubusercontent.com/58825866/78036834-60548e00-7330-11ea-9b4b-8b36984b2396.png)

Credential Engine is always the publisher.
The secondary publisher was added to the envelope in order to indicate a third party publisher. However, as the envelope is now managed by the Credential Registry, we could not directly update the property. Instead a third party organization would have to be added as publisher in the registry. We did not implement this approach. 

Regardless, I believe the publishedBy is a useful property to have in the payload. The search returns the payload, not the envelope. For the most part consumers will not be exposed to the envelope. 

In a third party relationship:
ownedBy, offeredBy: Wichita State
publishedBy: Kansas State Board of Education

**Comment by siuc-nate** [comment]
So far we have tried to avoid properties that only describe the record with respect to the registry (I think it's pretty much been limited to "last updated" in some cases?).  Technically, who put the record into the registry has no bearing on what the rest of the payload is describing, so it isn't really a property of that data.  

I could see doing this if we have a use case for consuming it and if we name and define the property very, very specifically as indicating which organization put the record into the Credential Registry.

Otherwise, we really should make use of the envelope for describing this kind of meta information - that's what it exists for.

**Comment by siuc-nate** [comment]
If the problem is that it's inconvenient to retrieve the envelope, then we should attack that problem instead of working around it.  I have reopened an old registry issue related to that ( https://github.com/CredentialEngine/CredentialRegistry/issues/146 ) with the suggestion that we enable retrieving an envelope by its CTID without needing to use the low level search workaround.

**Comment by mparsons-ce** [comment]
Using the search is not inconvenient. The same effort is required to use the search or the-soon-to-be-added-envelopes/{ctid} endpoint.
Part of the issue is how data is consumed. The /graph, /resources, and graph search don't return the envelope. If publishedBy and other proposed metadata is only on the envelope the user would have to know to use the /envelope endpoints for single requests (or do two requests).

**Comment by siuc-nate** [comment]
That's no different from the user having to know to get things like previous version data out of the envelope - it's meta information about the publishing, so it doesn't belong in the payload itself, in my opinion.

**Comment by siuc-nate** [comment]
Update:
- The ceterms:Pathway and ceterms:PathwaySet class now both have ceterms:ownedBy and ceterms:offeredBy (though it's not clear what "offering" a Pathway means), which should solve the original use case that @jeannekitchens mentions in the first post
- The Credential Registry now supports getting data by its publisher:
  - via [the Envelope](https://credentialengineregistry.org/envelopes/ce-c62c336f-a76d-4d7f-a777-367e7cc72d17)
  - via [internal processes](https://github.com/CredentialEngine/CredentialRegistry/issues/392) 
  - via [the Search API](https://github.com/CredentialEngine/CredentialRegistry/issues/364) 
- This issue hasn't come up in roughly a year

As such, I don't think we need to do anything here.  I am adding this to the next release's milestone tracking so we can evaluate it.

**Comment by jeannekitchens** [comment]
This is to float this issue back up to the top.  We have use cases for groups publishing pathways where the components are owned or offered by multiple organizations.  For example pathways that have Ivy Tech components and Southern New Hampshire University Components. We have a use case for pathways that have components offered and owned by multiple organizations that lead to a job offered by an employer.  The publishers isn't the "owner" and is one among many offerers and owrners.

**Comment by siuc-nate** [comment]
I'm afraid I'm not tracking - in your example, what would be solved by adding a "publishedBy" property?  It sounded like each of the entities already had ties to their respective organizations.

**Comment by jeannekitchens** [comment]
@stuartasutton @philbarker @siuc-nate I believe we've discussed this multiple times.  We need to close this or act upon it.  The point of this was to differentiate if, for example a pathway, pathway set, or other data that can include data from multiple contributors should indicate published by or source rather than offers or owns.  Case in point, a pathway can include components offered by mulitple organizations.   I thought we already closed this but see it's still open so this is to request input if we should not close this.

**Comment by stuartasutton** [comment]
Long thread!  Publishing, offering and owning are different actions and in some contexts have different legal consequences. I am for adding `publishedBy`. Phil at https://github.com/CredentialEngine/Schema-Development/issues/661#issuecomment-601355527 provides a definition.

**Comment by siuc-nate** [comment]
The registry envelope indicates the publisher of the resource to the registry. Is the objective here to also add that meta-information to the CTDL JSON itself? Or are we talking about the publisher (whatever that means) of the underlying resource (not the JSON in the registry)?  I don't think we want to mix infrastructure metadata about the record into the record data about the resource.
```
Envelope -[describes]-> JSON-LD Record -[describes]-> Real-world resource
```
To me, the notion of "published by" belongs (and is already present on) the Envelope.
